pallets/aura-ext/src/lib.rs

```diff
@@ -68,7 +77,32 @@ pub mod pallet {
 			// Fetch the authorities once to get them into the storage proof of the PoV.
 			Authorities::<T>::get();
 
-			T::DbWeight::get().reads_writes(2, 1)
+			let new_slot = Aura::<T>::current_slot();
+			let velocity = T::ConsensusHook::block_processing_velocity();
+
+			let (new_slot, authored) = match SlotInfo::<T>::get() {
+				Some((slot, authored)) if slot == new_slot => {
+					// New slot is the same as previous one, increment the
+					// the counter if within the limit.
+					if authored < velocity.get() {
```

Comment:
To match the issue description this should actually be
```rust
if authored <= velocity.get()
```

But it seems counter-intuitive to me. If there's a single core assigned to para, why would we allow `V + 1` blocks? This doesn't align with current behavior either.

@rphmeier

Reply:
Note that if this is implemented as a `pallet_parachain_system::ConsensusHook` with a fixed capacity, the rules we get allow authors to author N blocks, where N is the lesser of:
  * the amount of space remaining in the unincluded segment
  * V+1

There is more elaboration on the capacity/velocity duality in https://github.com/paritytech/cumulus/issues/2476 . But in effect, this is about building a backlog.

This _does_ align with current behavior as long as the unincluded segment capacity is 1. But this is a general piece of code for taking full advantage of asynchronous backing in Aura.